### PR TITLE
docs: fix jsdoc lint issues

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

Fix all jsdoc lint issues. In most cases, full documentation of all methods has added. In some cases invalid / partially complete jsdoc blocks for methods were removed if they added nothing to the readability of the code or converted to non jsdoc single line comments.

Generally speaking full documentation was added to all reducers. Components mostly  just had doc cleanup as per the above.

Reducer code was reorganised so hat all reducers follow the same basic structure and code order.

Some unused legacy code was removed.

## Motivation and Context:

Fix lint issues.
Improve DX

## How Has This Been Tested?

Manually

## Types of changes:

Docs updates

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [x] I have added tests to cover my changes where needed.
- [x] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: renderer/reducers/app.js [hunk 1]
```diff
@@ -41,27 +43,49 @@ export const TERMINATE_APP_SUCCESS = 'TERMINATE_APP_SUCCESS'
...
   return {
     type: SET_LOADING,
     isLoading,
   }
 }
 
+/**
+ * setLoading - Set app mount state.
+ *
+ * @param {boolean} isMounted Boolean indicating wether the app has been mounted
```
Comment (korhaliv) [inline renderer/reducers/app.js]: wether->whether

File: renderer/reducers/autopay.js [hunk 2]
```diff
@@ -37,6 +51,12 @@ export function enableAutopay(merchantId, limit) {
   }
 }
 
+/**
+ * disableAutopay - Disable autopay for a givem merchant.
```
Comment (korhaliv) [inline renderer/reducers/autopay.js]: `givem`

File: renderer/reducers/autopay.js [hunk 3]
```diff
@@ -71,25 +102,50 @@ export function initAutopay() {
...
   return {
     type: RESET_EDIT_MODE,
   }
 }
 
+/**
+ * openAutopayCreateModal - Open the autopay create modal for a given merchant.
+ *
+ * @param {string} merchantId Merchant Id
+ * @param {boolean} isEditMode Boolean indicating wether the modal is in edit mode
```
Comment (korhaliv) [inline renderer/reducers/autopay.js]: wether->whether

File: renderer/reducers/autopay.js [hunk 4]
```diff
@@ -101,15 +157,29 @@ export function openAutopayCreateModal(merchantId, isEditMode = false) {
...
+// ------------------------------------
+// Helpers
+// ------------------------------------
+
 /**
- * Adds or replaces (if it already exists) autopay entry
+ * addAutopayEntry - Add or replaces (if it already exists) an autopay entry.
+ *
+ * @param {object} state State data
+ * @param {{data}} data autopay configuration
```
Comment (korhaliv) [inline renderer/reducers/autopay.js]: a->A

File: renderer/reducers/autopay.js [hunk 5]
```diff
@@ -120,12 +190,26 @@ function addAutopayEntry(state, { data }) {
...
   const list = { ...state.list }
   delete list[merchantId]
   return { ...state, list }
 }
 
+/**
+ * setAutopayListFromArray - Populate the autopay list from a list of autopay configurations.
+ *
+ * @param {object} state State data
+ * @param {{list}} listList of autopay configurations
```
Comment (korhaliv) [inline renderer/reducers/autopay.js]: `listList`

File: renderer/reducers/autopay.js [hunk 6]
```diff
@@ -229,6 +315,14 @@ export { autopaySelectors }
 // ------------------------------------
 // Reducer
 // ------------------------------------
+
+/**
+ * autopayReducer - Autopay reducer.
+ *
+ * @param  {object} state = initialState Initial state
+ * @param  {object} action Action
+ * @returns {object} Final state
```
Comment (korhaliv) [inline renderer/reducers/autopay.js]: i'd say it returns `next` state and not final

File: renderer/reducers/contactsform.js [hunk 7]
```diff
@@ -1,22 +1,50 @@
...
 
+// ------------------------------------
+// Helpers
+// ------------------------------------
+
+/**
+ *fromSuggestedToRegular - Converts suggested nodes to a list of entries compatible with contact search results.
+ *
+ * @param {Array} suggestedNodes Suggested nodes list
+ * @returns {Array} search results compatible version of suggested nodes array
```
Comment (korhaliv) [inline renderer/reducers/contactsform.js]: s->S

File: renderer/reducers/modal.js [hunk 8]
```diff
@@ -32,13 +57,25 @@ export const openModal = (type, options) => (dispatch, getState) => {
   }
 }
 
+/**
+ * closeModal - Close a specific modal (or the top modal if id is ommitted).
+ *
+ * @param  {string} id Id  modal to close
```
Comment (korhaliv) [inline renderer/reducers/modal.js]: two spaces `Id  modal`

File: renderer/reducers/onboarding.js [hunk 9]
```diff
@@ -32,107 +76,212 @@ export const resetOnboarding = () => dispatch => {
...
   return {
     type: SET_AUTOPILOT,
     autopilot,
   }
 }
 
+/**
+ * setName - Set the blockchain to connect to.
+ *
+ * @param {'bitcoin'|'litecoin'} chain Chain name
```
Comment (korhaliv) [inline renderer/reducers/onboarding.js]: should be `{('bitcoin'|'litecoin')}` ?

File: renderer/reducers/onboarding.js [hunk 10]
```diff
@@ -32,107 +76,212 @@ export const resetOnboarding = () => dispatch => {
...
   return {
     type: SET_CHAIN,
     chain,
   }
 }
 
+/**
+ * setNetwork - Set the network to connect to.
+ *
+ * @param {'testnet'|'mainnet'} network Network name
```
Comment (korhaliv) [inline renderer/reducers/onboarding.js]: Should be `('testnet'|'mainnet')` ?

File: renderer/reducers/onboarding.js [hunk 11]
```diff
@@ -32,107 +76,212 @@ export const resetOnboarding = () => dispatch => {
...
   return {
     type: SET_PASSPHRASE,
     passphrase,
   }
 }
 
+/**
+ * setSeed - Set the seed.
+ *
+ * @param {Array} seed seed
```
Comment (korhaliv) [inline renderer/reducers/onboarding.js]: s->S

File: renderer/reducers/pay.js [hunk 12]
```diff
@@ -16,15 +34,42 @@ export const QUERY_ROUTES_FAILURE = 'QUERY_ROUTES_FAILURE'
...
+ */
+export const lightningPaymentUri = (event, { address }) => dispatch => {
+  dispatch(setRedirectPayReq({ address }))
+}
+
+/**
+ * bitcoinPaymentUri - Initiate bitcoin payment flow.
+ *
+ * @param  {event} event Event
+ * @param  {{ address, options }} options, decoded bip21 payment url
```
Comment (korhaliv) [inline renderer/reducers/pay.js]: `, decoded ` -> `Decoded`

File: renderer/reducers/pay.js [hunk 13]
```diff
@@ -42,42 +87,41 @@ export const queryFees = (address, amountInSats) => async (dispatch, getState) =
   }
 }
 
+/**
+ * queryRoutes - Find valid routes to make a payment to a node.
+ *
+ * @param {string} pubKey Destination node pubkey
+ * @param {number} amount desired amount in satoshis
```
Comment (korhaliv) [inline renderer/reducers/pay.js]: d->D

File: renderer/reducers/payment.js [hunk 14]
```diff
@@ -111,21 +174,51 @@ export const sendPayment = data => dispatch => {
...
 
+/**
+ * payInvoice - Pay a lightniung invoice.
+ * Controller code that wraps the send action and schedules automatic retries in the case of a failure.
+ *
+ * @param {object} options Options
+ * @param {string} options.payReq Payment request
+ * @param {number} options.amt Payment amount (in sats)
+ * @param {number} options.feeLimit The max fee to apply
+ * @param {boolean} options.isRetry Boolean indicating wether this is a retry attempt
```
Comment (korhaliv) [inline renderer/reducers/payment.js]: wether->whether

File: renderer/reducers/ticker.js [hunk 15]
```diff
@@ -36,39 +105,36 @@ export const setCryptoUnit = unit => async (dispatch, getState) => {
...
-
-export function recieveTickers({ rates }) {
-  return {
-    type: RECIEVE_TICKERS,
-    rates,
-  }
-}
-
+/**
+ * fetchTickers - Fetch all fiat tickers from currently active fiat tickert provider.
```
Comment (korhaliv) [inline renderer/reducers/ticker.js]: `tickert`

File: renderer/reducers/transaction.js [hunk 16]
```diff
@@ -96,6 +117,18 @@ export const receiveTransactions = ({ transactions }) => (dispatch, getState) =>
...
+/**
+ * sendCoins - Send an onchain transaction.
+ *
+ * @param  {object}  options Options
+ * @param  {number}  options.value Number of units to send
+ * @param  {string}  options.addr Destination address
+ * @param  {string}  options.cryptoUnit Crypto unit that value is donated in (will be converted to sats prior to send)
+ * @param  {number}  options.targetConf Number of blocks to target for conf time
+ * @param  {number}  options.satPerByte Sat per byte fee rate to apply
+ * @param  {boolean} options.isCoinSweep Boolean indicating wether this is a coin sweep (will send all funds).
```
Comment (korhaliv) [inline renderer/reducers/transaction.js]: wether->whether

File: renderer/reducers/transaction.js [hunk 17]
```diff
@@ -96,6 +117,18 @@ export const receiveTransactions = ({ transactions }) => (dispatch, getState) =>
   dispatch(fetchBalance())
 }
 
+/**
+ * sendCoins - Send an onchain transaction.
+ *
+ * @param  {object}  options Options
+ * @param  {number}  options.value Number of units to send
+ * @param  {string}  options.addr Destination address
+ * @param  {string}  options.cryptoUnit Crypto unit that value is donated in (will be converted to sats prior to send)
```
Comment (korhaliv) [inline renderer/reducers/transaction.js]: donated->denominated

File: electron/walletBackup/base/TokenBasedBackupService.js [hunk 18]
```diff
@@ -82,7 +80,7 @@ export default class TokenBasedBackupService extends EventEmitter {
    * isUsingTokens - This service is token based and requires tokens to operate
    * It also emits `tokensReceived` event.
    *
-   * @returns {boolean}
+   * @returns {boolean} Boolean indicating wether service uses tokens
```
Comment (korhaliv) [inline electron/walletBackup/base/TokenBasedBackupService.js]: wether->whether

File: renderer/components/UI/LndConnectionStringInput.js [hunk 19]
```diff
@@ -34,7 +24,7 @@ class LndConnectionStringInput extends React.Component {
   }
 
   /**
-   * Check for a valid lndconnect uri or BtcPayServer connection string.
+   * validate - Check for a valid lndconnect uri or BtcPayServer connection string.
    *
    * @param  {string}  value String to validate.
    * @returns {boolean}       Boolean indicating wether the string is a valid or not.
```
Comment (korhaliv) [inline renderer/components/UI/LndConnectionStringInput.js]: whether

**Comment by mrfelton** [comment]
Updated!

**Pull request merged**